core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java

```diff
@@ -374,4 +386,128 @@ static TableMetadata commit(TableOperations ops, UpdateTableRequest request) {
 
     return ops.current();
   }
+
+  private static BaseView baseView(View view) {
+    Preconditions.checkArgument(view instanceof BaseView, "View must be a BaseView");
```

Comment:
The Table methods use `IllegalStateException` with `Cannot wrap catalog that does not produce BaseTable`. Can you make the behavior similar?

Reply:
done